### DISCUSSION TITLE
Hotkeys in Town Selection Screen

### DISCUSSION THREAD

**Discussion by eos428** [ihhub/fheroes2#3438] [Ideas]
Hello all. Since we already have hotkeys implemented for splitting and some other places. I think we can look into adding hotkeys in town selection screen.

In which case I propose these hotkeys:
- B hotkey: Jump to build selection screen
- M hotkey: Jump to marketplace. Do note that H3HD's implementation has this hotkey as B. But personally, I like M more since I prefer B for build.
- S hotkey: Jump to mage guild
- W hotkey: Well (the building)
- Left and right keys OR up and down keys: Shift through towns we have 

Other things we can use our hotkeys with are probably shipyard and thieves guild.
What do you think? Anything else we'd like to add?

**Comment by LeHerosInconnu** [comment]
Hello @eos428,

Although I prefer to use the mouse... :D

- Maybe add the keys "1", "2", "3", "4", "5" and "6" on the keyboard and the numeric keypad to open the creature recruitment window of the corresponding levels of creatures.
- "T" key to open the Thieves' Guild screen.
- "O" key (for Ocean) or "N" key (for Navigation), to open the "Build a new boat" window (or use "G" key for Mage Guild and "S" key (for Shipyard)).
- "R" key (for Rumors) to open the Tavern window.

**Reply by eos428** [reply]
Hello @LeHerosInconnu 

Thanks a lot for the suggestions, most of them seem to be good. Only thing a bit iffy would be the shipyard hotkey, but we'll see
